src/devices/usbCamera/README.md

```diff
@@ -0,0 +1,287 @@
+
+## 2.1. PytonCameraHelper Cmake options
+
+The following CMake option should be used to compile UltraPython device:
+
+```
+ENABLE_yarpmod_usbCamera        ON
+ENABLE yarpmod_usbCameraRaw     ON
+```
+
+The only new options for UltraPython are:
+
+```
+YARP_USE_UDev                    ON
+COMPILE_WITHUNITTEST_ULTRAPYTH   OFF
+```
+
+UDev is in the advanced section. Unittest is OFF by default.
+
+## 2.2. Code formatting and naming convention
+
+The code formatting is done using the included `.clang-format` file.  
+The c++ style follows mainly the https://google.github.io/styleguide/cppguide.html with some notable exceptions.
+
+## 2.3. Unittest
+
+I have developed unit tests with the `gtest` and `gmock` library see https://google.github.io/googletest/.
+The tests can be found in `unittest` folder.
+
+# 3. Testing the video stream
+
+To test the stream I have developed the following procedure:
+
+- execute on **iCubHead** yarpserver
+- execute on **Xilinx** yarpdev
+- execute on **iCubHead** the following commands
+
+  ```
+  yarpdatadumper --name /log --rxTime --txTime --type image
+  yarp connect /grabber /log fast_tcp
+
+  ```
+
+- Analyze the data in `./log` using the Matlab scripts or manually, see above.
+
+## 3.1. Slow movment artifact
+
+These kind of artifacts are due to bad memory management. They should be checked by hand, looking at the singles frames.  
+<img src="img/artifacts.png" width="600px">
+
+## 3.2. Frames per second (FPS)
+
+Fps behaviour should be analyzed with the matlabscript in `matlabscript` folder, script name FPS_check.m.
+Once the script is executed choose the yarpdaadumper log.  
+Result example:  
+<img src="img/fps.png" width="600px">
+
+## 3.3. Development environment
+
+To develop the software on the Xilinx board is necessary to setup a remote development environment, as Xilinx board can't be used with a UI.  
+We have decided to use vscode with ssh extension.
+
+:exclamation:<u>To be done on iCub-head and with running Xilinx board.</u>
+
+Download and install vscode:https://code.visualstudio.com/  
+Install plugin for vscode named:
+
+- `ms-vscode-remote.remote-ssh`
+- `ms-vscode-remote.remote-ssh-edit`
+- `xaver.clang-format`
+
+Edit file ~/.ssh/config, add at the end:
+
+```
+Host Xilinx
+  HostName 10.0.0.233
+  User root
+  ForwardAgent yes
+```
+
+Connect using the correct host among your list (`Connect to Host in surrent windows`):
+
+<img src="img/dev001.png" width="500px">
+
+then you can open the remote folder on the same windows:
+
+<img src="img/dev002.png" width="500px">
+
+Choose the remote folder `/root/icubtech/yarp/src/devices/usbCamera/linux`
+
+A remote terminal is also available from the `Terminal` menu.
+
+:exclamation:_Troubleshooting_
+
+1. If vscode won't connect try to check Xilinx board file system.
+   `fsck / -y` Then restart boot Xilinx board and vscode.
+2. If vscode still won't connect try to delete, on Xilinx board, the following files:
+
+```
+rm /root/.vscode-server/.*
+```
```

Comment:
Super that you add the README! Could you please add also some lines for the generic usbCamera? The `ultraPython` will become a subsection of the documentation

Reply:
No problems for me, @pattacini all right for you?

Reply:
Yep